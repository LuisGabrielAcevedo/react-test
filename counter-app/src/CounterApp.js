import React, { useState } from "react";
import PropTypes from "prop-types";
import "./CounterApp.css";

const CounterApp = ({ value = 10 }) => {
  const [counter, setCounter] = useState(value);

  const handleAdd = () => setCounter(counter + 1);

  const handleSubtract = () => setCounter(counter - 1);

  const handleReset = () => setCounter(value);

  return (
    <>
      <h1>Counter App</h1>
      <p>{counter}</p>
      <button onClick={handleAdd}>+1</button>
      <button onClick={handleReset}>Reset</button>
      <button onClick={handleSubtract}>-1</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number,
};

export default CounterApp;
