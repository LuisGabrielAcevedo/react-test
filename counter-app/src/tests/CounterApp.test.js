import "@testing-library/jest-dom";
import React from "react";
import CounterApp from "../CounterApp";
import { shallow } from "enzyme";

describe("Pruebas en componente CounterApp.js", () => {
  let wrapper = shallow(<CounterApp />);

  beforeEach(() => {
    wrapper = shallow(<CounterApp />);
  });

  test("debe de mostrar <CounterApp/> correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar el value enviado por props", () => {
    const value = 100;
    const wrapper = shallow(<CounterApp value={value} />);
    const valueParrafo = wrapper.find("p").text();
    expect(valueParrafo).toBe(value.toString());
  });

  test("Debe de incrementar el contador", () => {
    wrapper.find("button").at(0).simulate("click");
    const valueParrafo = wrapper.find("p").text();
    expect(valueParrafo).toBe("11");
  });

  test("Debe de decrementar el contador", () => {
    wrapper.find("button").at(2).simulate("click");
    const valueParrafo = wrapper.find("p").text();
    expect(valueParrafo).toBe("9");
  });

  test("Debe de resetear el contador", () => {
    wrapper.find("button").at(2).simulate("click");
    wrapper.find("button").at(1).simulate("click");
    const valueParrafo = wrapper.find("p").text();
    expect(valueParrafo).toBe("10");
  });
});
