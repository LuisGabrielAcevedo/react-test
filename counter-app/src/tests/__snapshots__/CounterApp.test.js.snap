// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente CounterApp.js debe de mostrar <CounterApp/> correctamente 1`] = `
<Fragment>
  <h1>
    Counter App
  </h1>
  <p>
    10
  </p>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
